mws/apis/easyship.py

```diff
@@ -0,0 +1,90 @@
+from .. import utils
+
+
+class EasyShip(MWS):
+    """
+        Amazon EasyShip API
+        docs: http://docs.developer.amazonservices.com/en_IN/easy_ship/EasyShip_Overview.html
+    """
+    URI = "/EasyShip/2018-09-01"
+    VERSION = "2018-09-01"
+    NAMESPACE = '{https://mws.amazonservices.in/EasyShip/2018-09-01}'
+
+    def get_pickup_slots(self, marketplace_id=None, amazon_order_id=None, package_width=0,
+                         package_height=0, package_length=0, package_dimensions_uom='cm',
+                         package_weight=0, package_weight_uom='g'):
+        data = {
+            'Action': 'ListPickupSlots',
+            'MarketplaceId': marketplace_id,
+            'AmazonOrderId': amazon_order_id,
+            'PackageDimensions.Width': str(package_width),
+            'PackageDimensions.Length': str(package_length),
+            'PackageDimensions.Height': str(package_height),
+            'PackageDimensions.Unit': package_dimensions_uom,
+            'PackageWeight.Unit': package_weight_uom,
+            'PackageWeight.Value': str(package_weight)
+        }
+        return self.make_request(data)
+
+    def create_scheduled_package(self, marketplace_id=None, amazon_order_id=None, package_width=0,
+                                 package_height=0, package_length=0, package_dimensions_uom='cm',
+                                 package_weight=0, package_weight_uom='g',
+                                 package_slot_id=None, package_slot_start_time=None,
+                                 package_slot_end_time=None, package_identifier=None):
+        data = {
+            'Action': 'CreateScheduledPackage',
+            'MarketplaceId': marketplace_id,
+            'AmazonOrderId': amazon_order_id,
+            'PackageRequestDetails.PackageDimensions.Length': str(package_length),
+            'PackageRequestDetails.PackageDimensions.Width': str(package_width),
+            'PackageRequestDetails.PackageDimensions.Height': str(package_height),
+            'PackageRequestDetails.PackageDimensions.Unit': package_dimensions_uom,
+            'PackageRequestDetails.PackageWeight.Unit': package_weight_uom,
+            'PackageRequestDetails.PackageWeight.Value': str(package_weight),
+            'PackageRequestDetails.PackagePickupSlot.SlotId': package_slot_id,
+        }
+        if package_slot_start_time:
+            data.update({'PackageRequestDetails.PackagePickupSlot.PickupTimeStart': package_slot_start_time})
+        if package_slot_end_time:
+            data.update({'PackageRequestDetails.PackagePickupSlot.PickupTimeEnd': package_slot_end_time})
+        if package_identifier:
+            data.update({'PackageRequestDetails.PackageIdentifier': package_identifier})
+        return self.make_request(data)
+
+    def update_scheduled_package(self, marketplace_id=None, amazon_order_id=None, package_id=None,
+                                 slot_id=None, package_slot_start_time=None, package_slot_end_time=None):
```

Comment:
The [MWS docs for this method](https://docs.developer.amazonservices.com/en_IN/easy_ship/EasyShip_UpdateScheduledPackages.html) state:

> Currently you can specify only one ScheduledPackageUpdateDetails element.

I see you're passing single values for the contents of that one scheduled package; but I think it best if we do a little future-proofing by assuming that more than one can be sent eventually. That way, if they do decide to allow more than one package in the same request, we would already be covering it, without needing to change the code design and function signature.

I would say, then, we need to take an iterable of dicts, containing keys for `amazon_order_id`, `package_id`, `slot_id`, `slot_start_time`, and `slot_end_time`. The method code would just reprocess each dict to convert those keys to the "sub-keys" (`ScheduledPackageId.AmazonOrderId`, `PackagePickupSlot.SlotId`, etc.), and then `enumerate_keyed_params` that iterable.

Reply:
Additionally, the action name is `UpdateScheduledPackages`, which converted to `lowercase_with_underscores` should be `update_scheduled_packages` (`s` at the end). Please adjust that name to match.